label_studio/data_import/api.py

```diff
@@ -706,43 +708,79 @@ def get(self, *args, **kwargs):
 
 
 class DownloadStorageData(APIView):
-    """Check auth for nginx auth_request"""
+    """
+    Secure file download API for persistent storage (S3, GCS, Azure, etc.)
+
+    This view provides authenticated access to uploaded files and user avatars stored in
+    cloud storage or local filesystems. It supports two operational modes for optimal
+    performance and flexibility (simplicity).
+
+    ## Operation Modes:
+
+    ### 1. NGINX Mode (Default - USE_NGINX_FOR_UPLOADS=True)
+    - **High Performance**: Uses X-Accel-Redirect headers for efficient file serving
+    - **How it works**:
+      1. Validates user permissions and file access
+      2. Returns HttpResponse with X-Accel-Redirect header pointing to storage URL
+      3. NGINX intercepts and serves the file directly from storage
+    - **Benefits**: Reduces Django server load, better performance for large files
+
+    ### 2. Direct Mode (USE_NGINX_FOR_UPLOADS=False)
+    - **Direct Serving**: Django serves files using RangedFileResponse
+    - **How it works**:
+      1. Validates user permissions and file access
+      2. Opens file from storage and streams it with range request support
+      3. Supports partial content requests (HTTP 206)
+    - **Benefits**: Works without NGINX, supports range requests for media files
```

Comment:
> supports range requests for media files

I don't believe nginx doesn't now support range requests,maybe we need to configure something. 
This option, for example: https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_force_ranges